video/out/wayland_common.c

```diff
@@ -1978,6 +2499,19 @@ void vo_wayland_uninit(struct vo *vo)
     if (wl->registry)
         wl_registry_destroy(wl->registry);
 
+    if (wl->viewporter)
+      wp_viewporter_destroy (wl->viewporter);
+
+    if (wl->viewport)
+      wp_viewport_destroy (wl->viewport);
+
+    if (wl->video_viewport)
+      wp_viewport_destroy (wl->video_viewport);
+
+    if (wl->dmabuf)
+    	zwp_linux_dmabuf_v1_destroy (wl->dmabuf);
+    free(wl->drm_formats);
```

Comment:
I think this might be why I got an invalid free earlier when testing it. Should have an if check like the rest (also `talloc_free` should be used).

Reply:
C standard guarantees that `free` doesn't need a NULL check

Reply:
Ah yeah. Well `talloc_free` should solve the problem in any case.